clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -1533,12 +1533,17 @@ Expected<StringRef> clang(ArrayRef<StringRef> InputFiles, const ArgList &Args,
   };
 
   // Forward all of the `--offload-opt` and similar options to the device.
-  CmdArgs.push_back("-flto");
   for (auto &Arg : Args.filtered(OPT_offload_opt_eq_minus, OPT_mllvm))
     CmdArgs.append(
         {"-Xlinker",
          Args.MakeArgString("--plugin-opt=" + StringRef(Arg->getValue()))});
 
+  if (Triple.isNVPTX() || Triple.isAMDGPU()) {
```

Comment:
is there an upstream pr for this based on previous feedback?

Reply:
I thought we are going to do this after merging this. I will open one for that.

Reply:
Ok as long as you'll make an upstream PR lgtm to merge here first